lib/features/create_wager/create_wager_screen.dart

```diff
@@ -92,31 +98,35 @@ class CreateWagerScreen extends StatelessWidget {
                 Row(
                   children: [
                     Expanded(
-                      child: DropdownButtonFormField<String>(
-                        isExpanded: true,
-                        alignment: Alignment.center,
-                        decoration: InputDecoration(
-                          filled: true,
-                          fillColor: context.textBoxTextColor,
-                          border: OutlineInputBorder(
+                      child: GestureDetector(
+                        onTap: () => _showCategoryDialog(context),
+                        child: Container(
+                          padding: EdgeInsets.symmetric(
+                              vertical: 16, horizontal: 12),
+                          decoration: BoxDecoration(
                             borderRadius: BorderRadius.circular(8),
-                            borderSide: BorderSide.none,
+                            color: context.textBoxTextColor,
+                          ),
+                          child: Row(
+                            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                            children: [
+                              Text(
+                                selectedCategory ?? 'selectCategory'.tr(),
+                                style: TextStyle(color: context.blackColor),
```

Comment:
text color are definitely not black, check design properly, also there's a primaryTextColor for all texts